js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx

```diff
@@ -14,105 +15,139 @@ let fetchState: Record<
   }
 > = {};
 
-/**
- * Returns data from the indexedDB cache initially while loading is true.
- * Fetches data from the network/cache initially and does not receive any updates afterwards
- * Uses fetch-policy: no-cache to avoid slow apollo cache normalization
- */
+export class CacheManager<TQuery> {
+  private cache: ReturnType<typeof cache<string, CacheData<TQuery>>>;
+  private key: string;
+
+  constructor(key: string) {
+    this.key = `indexdbQueryCache:${key}`;
+    this.cache = cache<string, CacheData<TQuery>>({dbName: this.key, maxCount: 1});
+  }
+
+  async get(version: number): Promise<TQuery | null> {
+    if (await this.cache.has('cache')) {
+      const {value} = await this.cache.get('cache');
+      if (value && version === value.version) {
+        return value.data;
+      }
+    }
+    return null;
+  }
+
+  set(data: TQuery, version: number): Promise<void> {
+    return this.cache.set('cache', {data, version}, {expiry: new Date('3000-01-01')});
```

Comment:
should we be more conservative with expiry date ? what ensures these things get cleaned up?

Reply:
nit: if we do a far away date it should be 3030 (Deltron)